src/run_search.py

```
@@ -0,0 +1,192 @@
+def run_batch_search(
+    n_rules: int,
+    phase: ObservationPhase,
+    out_dir: Path,
+    steps: int = 200,
+    halt_window: int = 10,
+    base_rule_seed: int = 0,
+    base_sim_seed: int = 0,
+    world_config: WorldConfig | None = None,
+) -> list[SimulationResult]:
+    if n_rules < 1:
+        raise ValueError("n_rules must be >= 1")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)